pkg/manager/manager.go

```diff
@@ -77,6 +77,9 @@ type Manager interface {
 	// AddReadyzCheck allows you to add Readyz checker
 	AddReadyzCheck(name string, check healthz.Checker) error
 
+	// AddHook allows to add Runnables as hooks to modify the behavior of the controller.
+	AddHook(hook HookType, runnable Runnable) error
```

Comment:
```suggestion
	// Hook allows to register a Runnable as hooks to modify the behavior of the manager during specific events.
	Hook(hook HookType, runnable Runnable) error
```

Reply:
Accepting this outside of GitHub to also change the internal controller to implement the interface.